crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -62,6 +62,16 @@ def services
     end
   end
 
+  api :POST, "/api/upgrade/nodes", "Initiate the upgrade of all nodes"
+  api_version "2.0"
+  # This is gonna initiate the upgrade of all nodes.
+  # The method runs asynchronously, so there's a need to poll for the status and possible errors
+  def nodes
+    # FIXME: implement this as asynchronous call!
```

Comment:
ok, so there is no error that can happen in `Api::Upgrade.nodes` ? I mean do we need to check for its return value? `if true | false`